activeviam/atoti-ui-migration
# UI-8720 - Support migration of calculated measures to 5.10- content servers

## Description

Calculated measures on 5.11+ content servers are stored as JSON files. 5.10- content servers store them as XML files.
AUI 5.0 saves the calculated measures under `/ui/calculated_measures`, in their own format. AUI 5.1 stores them as regular measures under `pivot/entitlements/cm`.
Until now, we were assuming that AUI 5.1 calculated measures would always be stored as JSON files. Hence, when migrating calculated measures from AUI 5.0 to AUI 5.1, we were systematically creating them as JSON files under `pivot/entitlements/cm`. It's causing an issue when the content is stored on a 5.10- content server.

In this PR:
- we add the ability to add a `"contentServerVersion"` attribute at the root of the `servers.json` file
- if the indicated version there is below 5.11.0, the 5.0 calculated measures will be transformed and moved as XML files
- otherwise, they'll be transformed and moved as JSON files, like it is done now

## Test

- save a couple measures from a 5.0 UI
- use those measures in a dashboard that you save too
- download the content containing those measures and the saved dashboard from the content manager
- set the `"contentServerVersion"` attribute to `"5.10.12"` in the `servers.json` file
- apply the migration script
- make sure those those measures were migrated and they are in XML format
- try the same with a `"5.11.x"` content server version
- make sure the migrated measures are in JSON format
